e2e/suites/provider/cases/conjur/provider.go

```diff
@@ -44,43 +43,49 @@ const (
 	jwtK8sHostIDProviderName = "jwt-k8s-hostid-provider"
 )
 
-func newConjurProvider(f *framework.Framework) *conjurProvider {
+func newConjurProvider(f *framework.Framework, conjur *addon.Conjur) *conjurProvider {
 	prov := &conjurProvider{
 		framework: f,
+		addon:     conjur,
 	}
+
 	BeforeEach(prov.BeforeEach)
+
 	return prov
 }
 
 func (s *conjurProvider) CreateSecret(key string, val framework.SecretEntry) {
 	// Generate a policy file for the secret key
 	policy := createVariablePolicy(key, s.framework.Namespace.Name, val.Tags)
 
-	_, err := s.client.LoadPolicy(conjurapi.PolicyModePost, "root", strings.NewReader(policy))
+	_, err := s.addon.ConjurClient.LoadPolicy(conjurapi.PolicyModePost, "root", strings.NewReader(policy))
 	Expect(err).ToNot(HaveOccurred())
 
 	// Add the secret value
-	err = s.client.AddSecret(key, val.Value)
+	err = s.addon.ConjurClient.AddSecret(key, val.Value)
 	Expect(err).ToNot(HaveOccurred())
 }
 
 func (s *conjurProvider) DeleteSecret(key string) {
 	policy := deleteVariablePolicy(key)
-	_, err := s.client.LoadPolicy(conjurapi.PolicyModePatch, "root", strings.NewReader(policy))
+	_, err := s.addon.ConjurClient.LoadPolicy(conjurapi.PolicyModePatch, "root", strings.NewReader(policy))
 
 	Expect(err).ToNot(HaveOccurred())
 }
 
 func (s *conjurProvider) BeforeEach() {
-	ns := s.framework.Namespace.Name
-	c := addon.NewConjur(ns)
-	s.framework.Install(c)
-	s.client = c.ConjurClient
-	s.url = c.ConjurURL
-
-	s.CreateApiKeyStore(c, ns)
-	s.CreateJWTK8sStore(c, ns)
-	s.CreateJWTK8sHostIDStore(c, ns)
+	// setup test-app-sa policy
+	saName := "system:serviceaccount:" + s.framework.Namespace.Name + ":test-app-sa"
+	policy := createJwtHostPolicy(saName, "eso-tests")
```

Comment:
by now I have realized this entire framework is not cancellable so I guess this really is a separate issue :D